src/data/polynomial/high_order_derivative.lean

```diff
@@ -0,0 +1,153 @@
+import data.polynomial.derivative
+import algebra.ring.basic
+import data.polynomial.basic
+import logic.function.iterate
+import tactic.ring tactic.linarith
+
+noncomputable theory
+
+open finset nat polynomial
+open_locale big_operators
+
+namespace polynomial
+universes u
+variable {R : Type u}
+
+section semiring
+
+variables [semiring R] (f p q: polynomial R) (n k : ℕ)
+
+/--Defining the n-th derivative of polynomial f-/
+def deriv_n : polynomial R := derivative ^[n] f
```

Comment:
Bikeshed on the name?

The corresponding definition in the analysis library seems to be called `iterated_deriv`

Reply:
Let's wait and see if anyone has any other suggestion for names.

Reply:
I recommend the filename be changed to the name of this def, whatever it ends up being.

Reply:
I think it makes sense to have the same name as in analysis. Also +1 for Bryan's comment about filename.

Reply:
`iterated_deriv` in analysis isn't namespaced though, so I wonder if that will cause problems.

Reply:
I think `polynomial.iterated_deriv` should be fine. And we could even consider renaming `polynomial.derivative` to `polynomial.deriv`.

Reply:
I changed the name to `iterated_derivative` following `polynomial.derivative`